### ISSUE TITLE
elpa fails to build on macOS with clang and gfortran

### ISSUE THREAD

**Issue by davydden** [spack/spack#3878]
looking at the log, i have no idea what is going on here:
```
==> 'make'
fdep/fortran_dependencies.mk:76: warning: overriding commands for target `test/Fortran/elpa2_test_complex_api-test_complex2_api.use_mods_elpa2_test_complex_api.o'
fdep/fortran_dependencies.mk:76: warning: ignoring old commands for target `test/Fortran/elpa2_test_complex_api-test_complex2_api.use_mods_elpa2_test_complex_api.o'
fdep/fortran_dependencies.mk:76: warning: overriding commands for target `test/Fortran/elpa2_test_complex_api-test_complex2_api.def_mods_elpa2_test_complex_api.o'
fdep/fortran_dependencies.mk:76: warning: ignoring old commands for target `test/Fortran/elpa2_test_complex_api-test_complex2_api.def_mods_elpa2_test_complex_api.o'
fdep/fortran_dependencies.mk:117: target `/Users/davydden/spack/var/spack/stage/elpa-2016.05.004-k34djzvl2eyhob2wni3dcw6qict3xncf/elpa-2016.05.004/.fortran_dependencies/elpa2_test_complex_api' given more than once in the same rule.
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-am
fdep/fortran_dependencies.mk:76: warning: overriding commands for target `test/Fortran/elpa2_test_complex_api-test_complex2_api.use_mods_elpa2_test_complex_api.o'
fdep/fortran_dependencies.mk:76: warning: ignoring old commands for target `test/Fortran/elpa2_test_complex_api-test_complex2_api.use_mods_elpa2_test_complex_api.o'
fdep/fortran_dependencies.mk:76: warning: overriding commands for target `test/Fortran/elpa2_test_complex_api-test_complex2_api.def_mods_elpa2_test_complex_api.o'
fdep/fortran_dependencies.mk:76: warning: ignoring old commands for target `test/Fortran/elpa2_test_complex_api-test_complex2_api.def_mods_elpa2_test_complex_api.o'
fdep/fortran_dependencies.mk:91: /Users/davydden/spack/var/spack/stage/elpa-2016.05.004-k34djzvl2eyhob2wni3dcw6qict3xncf/elpa-2016.05.004/.fortran_dependencies/dependencies.mk: No such file or directory
fdep/fortran_dependencies.mk:117: target `/Users/davydden/spack/var/spack/stage/elpa-2016.05.004-k34djzvl2eyhob2wni3dcw6qict3xncf/elpa-2016.05.004/.fortran_dependencies/elpa2_test_complex_api' given more than once in the same rule.
  F90 DEPS /Users/davydden/spack/var/spack/stage/elpa-2016.05.004-k34djzvl2eyhob2wni3dcw6qict3xncf/elpa-2016.05.004/.fortran_dependencies/dependencies.mk
fdep/fortran_dependencies.mk:76: warning: overriding commands for target `test/Fortran/elpa2_test_complex_api-test_complex2_api.use_mods_elpa2_test_complex_api.o'
fdep/fortran_dependencies.mk:76: warning: ignoring old commands for target `test/Fortran/elpa2_test_complex_api-test_complex2_api.use_mods_elpa2_test_complex_api.o'
fdep/fortran_dependencies.mk:76: warning: overriding commands for target `test/Fortran/elpa2_test_complex_api-test_complex2_api.def_mods_elpa2_test_complex_api.o'
fdep/fortran_dependencies.mk:76: warning: ignoring old commands for target `test/Fortran/elpa2_test_complex_api-test_complex2_api.def_mods_elpa2_test_complex_api.o'
fdep/fortran_dependencies.mk:117: target `/Users/davydden/spack/var/spack/stage/elpa-2016.05.004-k34djzvl2eyhob2wni3dcw6qict3xncf/elpa-2016.05.004/.fortran_dependencies/elpa2_test_complex_api' given more than once in the same rule.
  PPFC     src/libelpa_public_la-elpa.lo
src/elpa.F90:58:6:

   use elpa1
      1
Fatal Error: Can't open module file ‘elpa1.mod’ for reading at (1): No such file or directory
compilation terminated.
make[1]: *** [src/libelpa_public_la-elpa.lo] Error 1
make: *** [all] Error 2
```
the closest issue i found online is https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=816086

**Comment by adamjstewart** [comment]
Try building in serial?

**Comment by davydden** [comment]
that's already a serial build :(( in parallel i have numerous of those `can't open module file...`.

**Comment by adamjstewart** [comment]
Hmm, I guess it needs to be _really_ serial then :P

**Comment by davydden** [comment]
i tried with the most recent `2017.05.002` version and there are still issues on macOS. However, I managed to nail it down to the auxiliary set of scripts [fdep](https://gitlab.mpcdf.mpg.de/elpa/elpa/tree/master/fdep). Those eventually should generate `<spack-build>/.fortran_dependencies/.dependencies.mk` file but it does not happen. If I copy-paste that file from `Ubuntu`, I can build `elpa`. 

Bottom line: something is fishy in those scripts and even adding
```
    depends_on('gmake@4.2.1:', type='build', when='platform=darwin')
    depends_on("autoconf", type='build', when='platform=darwin')
    depends_on("automake", type='build', when='platform=darwin')
```
does not help.

**Comment by davydden** [comment]
i don't thinks Spack can do anything about this issue, I notified developers several times but I don't think they care enough about macOS to fix the auxiliary scripts...

**Comment by boegel** [comment]
I saw this too on Linux, fixed for me with https://gitlab.mpcdf.mpg.de/elpa/elpa/commit/608cb9016a412834fc4cefec8ba483aa17137914
